release/release.gradle

```diff
@@ -242,5 +287,163 @@ artifacts {
+            if ( tag.endsWith( ".Final" ) ) {
+                tag = tag.replace( ".Final", "" )
+            }
+            logger.lifecycle( "Tagging '${tag}'..." )
+            executeGitCommand( 'tag', tag )
+        }
+
+        logger.lifecycle( "Adding commit to update version to '${project.developmentVersion}'..." )
+        project.ormVersionFile.text = "hibernateVersion=${project.developmentVersion}"
+        executeGitCommand( 'add', '.')
+        executeGitCommand( 'commit', '-m', project.developmentVersion )
+
+        if ( tag != null ) {
+            logger.lifecycle("Pushing branch and tag to remote '${project.gitRemote}'...")
+            executeGitCommand( 'push', '--atomic', project.gitRemote , project.gitBranch, tag )
+        }
+        else {
+            logger.lifecycle("Pushing branch to remote '${project.gitRemote}'...")
+            executeGitCommand( 'push', project.gitRemote , project.gitBranch )
+        }
+    }
+}
+
+static String executeGitCommand(Object ... subcommand){
+    List<Object> command = ['git']
+    Collections.addAll( command, subcommand )
+    def proc = command.execute()
+    def code = proc.waitFor()
+    def stdout = inputStreamToString( proc.getInputStream() )
+    def stderr = inputStreamToString( proc.getErrorStream() )
+    if ( code != 0 ) {
+        throw new GradleException( "An error occurred while executing " + command + "\n\nstdout:\n" + stdout + "\n\nstderr:\n" + stderr )
+    }
+    return stdout
+}
+
+static String inputStreamToString(InputStream inputStream) {
+    inputStream.withCloseable { ins ->
+        new BufferedInputStream(ins).withCloseable { bis ->
+            new ByteArrayOutputStream().withCloseable { buf ->
+                int result = bis.read();
+                while (result != -1) {
+                    buf.write((byte) result);
+                    result = bis.read();
+                }
+                return buf.toString( StandardCharsets.UTF_8.name());
+            }
+        }
+    }
+}
+
+class ChangeLogFile {
+
+    // Get the Release Notes from Jira and add them to the Hibernate changelog.txt file
+    static void update(String releaseVersion) {
+        def text = ""
+        File changelog = new File( "changelog.txt" )
+        def newReleaseNoteBlock = getNewReleaseNoteBlock(releaseVersion)
+        changelog.eachLine {
+            line ->
+                if ( line.startsWith( "Note:" ) ) {
+                    text += line + System.lineSeparator() + System.lineSeparator() + newReleaseNoteBlock
+                }
+                else {
+                    text += line + System.lineSeparator()
+                }
+        }
+        changelog.text = text
+    }
+
+    // Get the Release Notes from Jira
+    static String getNewReleaseNoteBlock(String releaseVersion) {
+        def restReleaseVersion;
+        if ( releaseVersion.endsWith( ".Final" ) ) {
+            restReleaseVersion = releaseVersion.replace( ".Final", "" )
```

Comment:
The above code makes the assumption that `.Final` appears only once, which is a sound assumption. But
```
restReleaseVersion = releaseVersion.substring( 0, releaseVersion.lastIndexOf(".Final") )
```
would be more robust and accurate.

Reply:
I think there is a perfect solution in Groovy in one-liner as
```
def restReleaseVersion = releaseVersion.replaceFirst( /\.Final$/, '' )
```
I tested it at https://groovyconsole.appspot.com/